### PR TITLE
🐛 Handle surrounding newlines and spaces in user names and passwords

### PR DESCRIPTION
**Pull request by dtantsur**
It's a common error to include a newline in secrets with user names and
passwords. So common, that I made it in my metal3-dev-env patch.

Log both values on user name mismatch.

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

File: pkg/ironic/secrets.go [hunk 1]
```diff
@@ -45,10 +45,14 @@ func generatePassword() ([]byte, error) {
 	return password, nil
 }
 
-func generateHtpasswd(user, password string) (string, error) {
+func normalizeSecretValue(value []byte) string {
+	return strings.Trim(string(value), " \n\r")
```
Comment (tuminoid) [inline pkg/ironic/secrets.go:49]: While at it, maybe add `\t` to be trimmed out as well?

Comment (dtantsur) [reply pkg/ironic/secrets.go:49]: A good idea

File: pkg/ironic/secrets_test.go [hunk 2]
```diff
@@ -52,12 +52,32 @@ func TestIsValidUser(t *testing.T) {
...
+			Password: "pa$$w0rd\n",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Scenario, func(t *testing.T) {
+			result, err := generateHtpasswd([]byte(tc.User), []byte(tc.Password))
+			assert.NoError(t, err)
+			user, password, ok := strings.Cut(result, ":")
+			assert.Truef(t, ok, "%s does is not separated with a colon", result)
```
Comment (tuminoid) [inline pkg/ironic/secrets_test.go:76]:
```suggestion
			assert.Truef(t, ok, "%s is not separated with a colon", result)
```
Missed this earlier, funny typo.

**Review by tuminoid — COMMENTED** [review]
/approve

**Pull request merged**